File: python/code/5-sinusoidal-wave.py
import numpy as np
import matplotlib.pyplot as plt

# Sampling frequency
fs = 1000

# Time
t = np.arange(0, 1, 1 / fs)

# Frequencies
f1 = 5
f2 = 10
f3 = 20

# Sinusoidal signals
x1 = np.sin(2 * np.pi * f1 * t)
x2 = np.sin(2 * np.pi * f2 * t)
x3 = np.sin(2 * np.pi * f3 * t)

# Plot
plt.figure(figsize=(12, 8))

plt.subplot(3, 1, 1)
plt.plot(t, x1)
plt.title("5 Hz Sinusoidal Wave")
plt.xlabel("Time (s)")
plt.ylabel("Amplitude")
plt.grid(True)

plt.subplot(3, 1, 2)
plt.plot(t, x2)
plt.title("10 Hz Sinusoidal Wave")
plt.xlabel("Time (s)")
plt.ylabel("Amplitude")
plt.grid(True)

plt.subplot(3, 1, 3)
plt.plot(t, x3)
plt.title("20 Hz Sinusoidal Wave")
plt.xlabel("Time (s)")
plt.ylabel("Amplitude")
plt.grid(True)

plt.tight_layout()
plt.show()

File: matlab/code/5-sinusoidal-wave.m
clc;
clear;
close all;

% Sampling frequency
Fs = 1000;

% Time vector
t = 0:1/Fs:1;

% Different frequencies
f1 = 5;
f2 = 10;
f3 = 20;

% Generate sinusoidal signals
x1 = sin(2*pi*f1*t);
x2 = sin(2*pi*f2*t);
x3 = sin(2*pi*f3*t);

% Plot 5 Hz
figure;
plot(t, x1);
grid on;
xlabel('Time (s)');
ylabel('Amplitude');
title('Sinusoidal Wave - 5 Hz');
xlim([0 1]);

% Plot 10 Hz
figure;
plot(t, x2);
grid on;
xlabel('Time (s)');
ylabel('Amplitude');
title('Sinusoidal Wave - 10 Hz');
xlim([0 1]);

% Plot 20 Hz
figure;
plot(t, x3);
grid on;
xlabel('Time (s)');
ylabel('Amplitude');
title('Sinusoidal Wave - 20 Hz');
xlim([0 1]);